brave/brave-core
# Add NFT grid

## Resolves https://github.com/brave/brave-browser/issues/29274 

⚠️ **Note:** check the [main umbrella issue](https://github.com/brave/brave-browser/issues/29291) first, to have an overview of what's been addressed in this PR and what is out of scope.

## Details

Implements new NFT tab in the Wallet section.
The layout will use an auto-resizing grid that adapts the number of columns depending on the width of the device screen.
The logic that handles the calculation of the span count is implemented in `AutoFitVerticalGridLayoutManager`.

The new layout follows the Figma design provided and will default to "All networks" when multi-chain support will be merged.

### Smartphone

![Screenshot_1679593935](https://user-images.githubusercontent.com/3308503/227300183-42ee95bc-4460-460a-92e9-d08ca862f965.png)

![Screenshot_1679593945](https://user-images.githubusercontent.com/3308503/227300265-23d2ae74-1ded-4f0a-8e94-0e60109e377b.png)

### Tablet

![Screenshot_1679591431](https://user-images.githubusercontent.com/3308503/227300675-99a3cbb0-c4e4-443b-a1cd-9cea8dcb7a67.png)

![Screenshot_1679591440](https://user-images.githubusercontent.com/3308503/227300726-4e9e5080-4c4f-4ed7-bda4-bec3af798696.png)

## Empty Grid

When no NFTs have been added, shows a different layout. 
Implemented following given Figma design.


![Screenshot_1679930599](https://user-images.githubusercontent.com/3308503/227987084-60ab3504-b397-483d-b149-ac845b35675a.png)


![Screenshot_1679930590](https://user-images.githubusercontent.com/3308503/227987134-f21d6b91-dfb9-4644-b884-4c633deb50b4.png)





## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Note for QA

Make sure to test against the latest Figma design as the layout has been updated quite consistently (e.g. card view has been removed).

## Test Plan:

- Open Wallet section
- If not already selected, choose "Solana Mainnet Beta"
- Tap on "NFTs" tab
- Tap on "Edit visible NFTs" then "Add custom asset"
- Select the NFT checkbox
- Add token address of a Solana NFT (e.g. [C2b9dQ68jFRTX1xuMedq6FfKCjTWvdWnu8DPQC1zbmZq](https://opensea.io/assets/solana/C2b9dQ68jFRTX1xuMedq6FfKCjTWvdWnu8DPQC1zbmZq))
- Populate token name and token symbol
- Tap on "Add"
- Tap on "Save account watchlist"
- Observe the NFT appears in the new section
- Tap on the newly added Solana NFT and open NFT detail screen
- Try to add more Solana NFTs and observe they appear correctly in the grid view
